src/uu/tsort/src/tsort.rs

```diff
@@ -63,11 +63,17 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
 
     // Create the directed graph from pairs of tokens in the input data.
     let mut g = Graph::new(input.to_string_lossy().to_string());
-    for ab in data.split_whitespace().collect::<Vec<&str>>().chunks(2) {
-        match ab {
-            [a, b] => g.add_edge(a, b),
-            _ => return Err(TsortError::NumTokensOdd(input.to_string_lossy().to_string()).into()),
-        }
+
+    let mut edge_tokens = data.split_whitespace();
```

Comment:
maybe document this a bit more to explain what it is doing ;)

Reply:
Expanded a bit